src/hotspot/share/gc/g1/g1FullCollector.cpp

```diff
@@ -322,6 +330,10 @@ void G1FullCollector::phase2_prepare_compaction() {
 
   phase2a_determine_worklists();
 
+  if (!has_compaction_targets()) {
+    return;
```

Comment:
Maybe it is useful to add a log message here that no compaction targets were found before exiting. Something like:
```
log_debug(gc, phases) debug("Phase 2: No compaction targetes");
```
So that people looking at logs have a better understanding why the other phases are skipped and the next log message will be about phase 5.

Reply:
Wouldn't `log_info(gc, phases) ("No Regions selected for compaction.` above (on the info level) already cover this?

Reply:
It is fine for me either way, can keep both or leave just the log_info.

Reply:
The `log_info` one is sufficient.